onnxruntime/core/common/path.cc

```diff
@@ -0,0 +1,293 @@
+// Copyright (c) Microsoft Corporation. All rights reserved.
```

Comment:
A lot of the code in path.h/cc is kinda reinventing the wheel. Is it possible to use the Windows/Posix APIs to do the same? Even better would've been to use boost::filesystem, but I guess we'll need to evaluate how much bloat it brings.

Reply:
std::filesystem in C++17 would be great! i was not aware we could use boost.
Windows/POSIX APIs are usable, but in my experience not as friendly or cross platform.

Reply:
I think we can allow all the windows build use C++17.